### PR TITLE
Feat: decidim private fields

### PR DESCRIPTION
**Pull request by froger**
Include a new feature in the Awesome module: Add private fields to custom proposals.

As a participation designer, I can define private fields for proposals to ask proposers for personal information. This personal information could be required to submit a proposal and should not involve an extra step in the proposal submission process.

**Context**
The city of Lausanne wanted to request private fields for their Participatory Budgeting (PB) in order to:
- Get consent for contacting proposers.
- Obtain a phone number.
- Collect additional information on the project (e.g., postal addresses to send materials).

Open Source Politics also contributes, financing adaptations for export and compatibility with version 0.27. <3

**Solution**

In the custom field administration, add another "custom field" zone, reserved for private fields.
In the form, add a section at the end for private fields.
Add an exporter, not exposed to open-data, to export personal data from the administration.
Change exporters of proposals to include custom fields.

**Screenshots**
_Proposal form, user side_
screencapture-theme-dev-voca-city-processes-nisi-non-f-14-proposals-45-edit-draft-2024-05-14-11_25_27

_Confirm screen, without private data_
Screenshot 2024-05-14 at 11 26 27

_Published proposal, without private data_
Screenshot 2024-05-14 at 11 26 39

_Visualization of the fields, admin side_
Screenshot 2024-05-14 at 11 34 47

_Export options for private data_
Screenshot 2024-05-14 at 11 31 22

_Proposal administration_
screencapture-theme-dev-voca-city-admin-decidim-awesome-config-proposal-custom-fields-2024-05-14-10_15_08

### CODE REVIEW

**Comment by froger** [comment]
So from the pipelines, I see I need to do at least: 

- [ ] Fix tests to have pipelines passing
- [ ] Fix linting

File: db/migrate/20230309124413_create_private_proposal_fields.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreatePrivateProposalFields < ActiveRecord::Migration[5.2]
+  def change
+    create_table :decidim_awesome_private_proposal_fields do |t|
+      t.text :private_body, default: '<xml></xml>'
```
[the remark this answers is not in the document]
Comment (microstudi) [reply db/migrate/20230309124413_create_private_proposal_fields.rb:6]: Also, we should encrypt this field if it's going to keep private data

**Review by microstudi — COMMENTED** [review]
Hi, thanks for this. It is interesting but I think we should change some things.
The most important is that we already have a table for extra things related to a proposal (see the comments). we should use that table.

Another is that I think we should do this for the 28 series (branch `develop` for the moment), as this is going to be to much work to maintain in both parts.

Then, shouldn't we use a different menu for this private fields? i think a low percentage of customization are going to use private fields and rendering duplicated FormRenders in every box might slow down the browser quite a lot.

For overrides, we are using deface, this way we can touch smaller parts of views and it makes easier to maintain.

And finally, another thought that comes to mind is that we should use the encrypt built in in some attributes in Decidim (like authorizations). Precisely because it uses private data.

**Comment by froger** [comment]
Putting this PR in draft the time I make the necessary updates.

**Comment by simonaszilinskas** [comment]
Hey @froger , just wanted to ask if you had any news concerning the requested changes

**Comment by froger** [comment]
Hello all, 

* The administration looks now better, but still is without tabs. After discussing with Lucien, we will make a design proposal to refactor the administration of decidim awesome in a more sustainable way. 
* The private fields are now encrypted
* The PR should looks a bit better, synced with develop branch
* private fields are now a ` jsonb` , that can handle translation (still don't know why we would translate private data) 

We still miss: 
- pipelines that should pass
- tests should pass, it seems some tests on collaboratives drafts are not passing, didn't checked this out yet.

**Comment by froger** [comment]
> > Hey @froger , just wanted to ask if you had any news concerning the requested changes
> 
> also, this feature will may be cover with https://github.com/orgs/decidim/projects/24/views/12?pane=issue&itemId=49965362.

Our customer needs where quiet specifics to having a "private section" of the form. I am unsure the feature you mention will attend their need, but we can discuss on this.

**Comment by microstudi** [comment]
@froger if you are ok with it, we are going to finish this PR as we have time and certain pressure to finish it this month.

I'd like very much to maintain you as the author of the PR to recognize your good work. To do so, you should create this PR from a branch in this repository (I've invited you now), then we will be able to add commits to it in order to finish it.

On the other hand, we will make a new PR for the develop branch to add this feature to the 0.28 too.

**Comment by froger** [comment]
closing in favor of #314
@microstudi as discussed, I've open a branch local to this repository. Feel free to update the code ;)
Thanks a lot for your help.

**Comment by microstudi** [comment]
Thanks to you @froger , we'll finish this!

**Pull request closed without merging**